### PR TITLE
Fixed 'Bug 58551 - Memory leak when editing .xaml files'

### PR DESCRIPTION
**Pull request by mkrueger**
Disposing the editor would be enough to fix that leak. But there is no
reason a projection should listen for file changes since projected
files are purely virtual and change only when the main document
changes. Giving the text editor type to the backends can allow some
optimizations for non UI text editor use cases.

### CODE REVIEW

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs [hunk 1]
```diff
@@ -3625,5 +3636,7 @@ Task<ScopeStack> ITextEditorImpl.GetScopeStackAsync (int offset, CancellationTok
 				return this.TextEditor.HasFocus;
 			}
 		}
+
+		public TextEditorType TextEditorType1 { get => textEditorType; set => textEditorType = value; }
```
Comment (mrward) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs]: Do we need this TextEditorType1 property? I cannot see it used anywhere else.

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/EditorFactory.cs [hunk 2]
```diff
@@ -69,18 +69,33 @@ IReadonlyTextDocument ITextEditorFactory.CreateNewReadonlyDocument (MonoDevelop.
...
+		}
+
+		ITextEditorImpl ITextEditorFactory.CreateNewEditor (IReadonlyTextDocument document)
 		{
 			return new SourceEditorView (document);
 		}
 
+		ITextEditorImpl ITextEditorFactory.CreateNewEditor (TextEditorType textEditorType)
+		{
+			return new SourceEditorView ( textEditorType);
```
Comment (KirillOsenkov) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/EditorFactory.cs]: >  [](start = 32, length = 1)

no space here

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs [hunk 3]
```diff
@@ -175,15 +175,26 @@ void ResetRemoveMarker ()
...
+		TextEditorType textEditorType;
+
+		public TextEditorType TextEditorType {
+			get {
+				return textEditorType;
+			}
+		}
 
-		public SourceEditorView(string fileName, string mimeType)
+		public SourceEditorView (TextEditorType textEditorType = TextEditorType.Default)
```
Comment (KirillOsenkov) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs]: >textEditorType [](start = 42, length = 14)

is this being used?

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs [hunk 4]
```diff
@@ -175,15 +175,26 @@ void ResetRemoveMarker ()
 			});
 		}
 
-		bool loadedInCtor = false;
+		bool loadedInCtor = false;
+		TextEditorType textEditorType;
```
Comment (KirillOsenkov) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs]: >textEditorType [](start = 17, length = 14)

is this ever assigned?

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs]: That thing was auto generated - strange didn't see where it did go thought it wasn't working :/. It's a really bad habbit to generate code 'somewhere' in a large file. Our old insertion cursor mode was a good way to tell the user what gets inserted and where.

It's sad that we lose this :(

**Pull request merged**